packages/melange.ppx/ppx.ml

```diff
@@ -401,6 +544,20 @@ module Debug = struct
 end
 
 let () =
-  Driver.register_transformation ~impl:structure_mapper
+  Driver.add_arg "-bundler"
```

Comment:
at first I didn't like the idea of coupling our melange_ppx with a bundler, and I would prefer to keep a `--assets-hash-fn` where it specifies "xxhash64" or even a format like webpack, but after seeing the complexity it carries... It's perfectly fine to keep it coupled.

Did you thought the same?

Reply:
It's a bit tricky, because the behavior changes a bit between bundlers:

- webpack is the most flexible of those i've investigated, allowing to use [any kind of hash function](https://webpack.js.org/configuration/output/#outputhashfunction) and [any kind of template](https://webpack.js.org/guides/asset-modules/#custom-output-filename) to build the resulting asset name
- rspack allows [any kind of template](https://www.rspack.dev/config/output#outputassetmodulefilename), but hash functions are delimited to xxhash64 or md4 (and the hashes are not really from the file content, unlike webpack)
- esbuild [allows any template](https://esbuild.github.io/api/#asset-names), but only supports xxhash64 (postprocessed with base32, which makes it different from the two above)

How do you think we should model this? It certainly seems we should allow users to configure somewhere the file name template to be used to generate the final paths, right? But we should somehow guarantee that the templates are also compatible across bundlers?

In my mind, this wasn't blocking an initial mvp of this feature because users can always update their JS bundler config to use the default behavior of each bundler (which is what is currently implemented). Wdyt?